### PR TITLE
CM Pxe distribution point

### PR DESCRIPTION
**Pull request by jeffotterpohl**
#### Task list

- [X ] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [X ] Resource documentation added/updated in README.md.
- [X ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [X] Comment-based help added/updated.
- [X] Localization strings added/updated in all localization files as appropriate.
- [X] Examples appropriately added/updated.
- [X] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [X] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/42)

### CODE REVIEW

File: source/DSCResources/DSC_CMPxeDistributionPoint/DSC_CMPxeDistributionPoint.psm1 [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+            throw ($script:localizedData.DistroPointInstall -f $SiteServerName)
+        }
+
+        if ((($PSBoundParameters.EnablePxe -eq $false) -or ([string]::IsNullOrEmpty($PSBoundParameters.EnablePxe) -and
+            $state.EnablePxe -eq $false)) -and (($PSBoundParameters.EnableNonWdsPxe) -or
+            ($PSBoundParameters.EnableUnknownComputerSupport) -or ($PSBoundParameters.AllowPxeResponse) -or
+            ($PSBoundParameters.PxeServerResponseDelaySec) -or ($PSBoundParameters.UserDeviceAffinity) -or
+            ($PSBoundParameters.PxePassword)))
+        {
+            throw $script:localizedData.PxeThrow
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMPxeDistributionPoint/DSC_CMPxeDistributionPoint.psm1:171]: Do we want a warning in Test for this? Same for NonWdsThrow?

File: source/DSCResources/DSC_CMPxeDistributionPoint/DSC_CMPxeDistributionPoint.schema.mof [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[ClassVersion("1.0.0"), FriendlyName("CMPxeDistributionPoint")]
+class DSC_CMPxeDistributionPoint : OMI_BaseResource
+{
+    [Key, Description("Specifies the SiteCode for the Configuration Manager site.")] String SiteCode;
+    [Key, Description("Specifies the SiteServer to install the role on.")] String SiteServerName;
```
Comment (rchristman89) [inline source/DSCResources/DSC_CMPxeDistributionPoint/DSC_CMPxeDistributionPoint.schema.mof:5]: Is this a role that is being installed?

**Comment by NEllis280** [comment]
Tested in the lab and everything works as expected.

**Comment by NEllis280** [comment]
Once the other comments are resolved, I'm good with this PR.

**Comment by jeffotterpohl** [comment]
Being able to specify a certain network adapter for PXE would be good feature request.

**Pull request merged**